src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs

```diff
@@ -85,7 +90,16 @@ internal static void Return(this OtlpCollector.ExportTraceServiceRequest request
 
             foreach (var scope in resourceSpans.ScopeSpans)
             {
-                RepeatedFieldOfSpanSetCountAction(scope.Spans, 0);
+                try
```

Comment:
Do we need a try-catch block? It's either going to always find the field `count` or never find it. This would be confirmed in the static field initialization of `repeatedFieldOfSpanSetCountAction`, right?

Reply:
> Do we need a try-catch block?

_Maybe_?

I was intentionally cautious here because just though there's a `count` field doesn't mean that the IL we've emitted to reset it will be valid in the future. Maybe it's unlikely, but imagine if the type of the `count` field were to change? Might not break anything, but is it worth risking it?

Reply:
I guess it's probably impossible to guarantee any kind of safety here. This reflection hack is definitely danger zone territory.

I just did a little experiment simulating changing the type of a count field on a class:

```csharp
using System.Reflection;
using System.Reflection.Emit;

var setFieldAction = SetFieldAction<ClassWithPrivateCountField>("count");
var obj = new ClassWithPrivateCountField(10);
Console.WriteLine(obj.Count);
setFieldAction(obj, 0);

// This line throws a NullReferenceException because the count field was set to null
Console.WriteLine(obj.Count);

Action<T, int> SetFieldAction<T>(string fieldName)
{
    var field = typeof(T).GetField(fieldName, BindingFlags.NonPublic | BindingFlags.Instance);

    DynamicMethod dynamicMethod = new DynamicMethod(
        "CreateSetCountAction",
        null,
        new[] { typeof(T), typeof(int) },
        typeof(ClassWithPrivateCountField).Module,
        skipVisibility: true);

    var generator = dynamicMethod.GetILGenerator();

    generator.Emit(OpCodes.Ldarg_0);
    generator.Emit(OpCodes.Ldarg_1);
    generator.Emit(OpCodes.Stfld, field);
    generator.Emit(OpCodes.Ret);

    return (Action<T, int>)dynamicMethod.CreateDelegate(typeof(Action<T, int>));
}                       

class ClassWithPrivateCountField
{
    private AnotherClass count;
    public int Count => count.Count;
    public ClassWithPrivateCountField(int count)
    {
        this.count = new AnotherClass(count);
    }
}

class AnotherClass
{
    private int count;
    public int Count => count;
    public AnotherClass(int count)
    {
        this.count = count;
    }
}
```

Another risk of manipulating internal state of a class is that the behavior of the class may change the way the state is used.

Reply:
I'm open to folks opinion that any change that would break this hack for us in the future is highly unlikely. In which case we could just leave the hack be.

Reply:
I guess it's a good idea to be more defensive then.

Reply:
@alanwest What you could do in `CreateRepeatedFieldOfMetricSetCountAction` is after you create the `Action<RepeatedField<OtlpMetrics.Metric>` delegate test it on an instance you new up. If it works, that means we are good to use the generated code. If it throws you could return null to throw it away and go into fallback mode. What that would do is save the try/catch perf hit on each item exported. Basically test it once on init to verify it is good before going hot.